lib/nx_deflate.c

```diff
@@ -650,6 +647,15 @@ int nx_deflateInit2_(z_streamp strm, int level, int method, int windowBits,
 	else
 		s->strategy = Z_DEFAULT_STRATEGY;
 
+	if (level < 5 || level > 9)
```

Comment:
I think we shouldn't accept `(level < 0 || level > 9)`.

Reply:
Agreed. I replaced the ifs with a switch-case. Now all these values whould return `Z_STREAM_ERROR` like zlib does.